VirtoCommerce.Platform.Data/ExportImport/PlatformExportImportManager.cs

```diff
@@ -128,119 +126,250 @@ public async Task ImportAsync(Stream stream, PlatformExportManifest manifest, Ac
         private async Task ImportPlatformEntriesInternalAsync(ZipArchive zipArchive, PlatformExportManifest manifest, Action<ExportImportProgressInfo> progressCallback, ICancellationToken cancellationToken)
         {
             var progressInfo = new ExportImportProgressInfo();
+            var jsonSerializer = GetJsonSerializer();
+            var batchSize = 20;
 
             var platformZipEntries = zipArchive.GetEntry(PlatformZipEntryName);
             if (platformZipEntries != null)
             {
-                PlatformExportEntries platformEntries;
                 using (var stream = platformZipEntries.Open())
                 {
-                    platformEntries = stream.DeserializeJson<PlatformExportEntries>(GetJsonSerializer());
+                    using (var streamReader = new StreamReader(stream))
+                    using (var reader = new JsonTextReader(streamReader))
+                    {
+                        while (reader.Read())
+                        {
+                            if (reader.TokenType == JsonToken.PropertyName)
+                            {
+                                if (manifest.HandleSecurity && reader.Value.ToString().EqualsInvariant("Roles"))
+                                {
+
+                                    await reader.DeserializeJsonArrayWithPagingAsync<Role>(jsonSerializer, batchSize, async items =>
+                                        {
+                                            foreach (var role in items)
+                                            {
+                                                if (await _roleManager.RoleExistsAsync(role.Name))
+                                                {
+                                                    await _roleManager.UpdateAsync(role);
+                                                }
+                                                else
+                                                {
+                                                    await _roleManager.CreateAsync(role);
+                                                }
+
+                                                var roleExist = await _roleManager.FindByNameAsync(role.Name);
+                                                var permissions = await _roleManager.GetClaimsAsync(roleExist);
+
```

Comment:
During import role permissions aren't synchronized, only new permissions will bee added, need to also handle deleted permissions

Reply:
Claims are updated when call CustomRoleManager.UpdateAsync. There is Permissions in the Patch.
have checked it.
So, removed the code from PlatformExportImportManager.cs